bin/reth/src/commands/poa/mod.rs

```
@@ -103,6 +105,57 @@ use tokio::{
 };
 use tracing::{debug, error, info};
 
+use reth_fs_util;
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct UtxosRecoveryConfig {
+    utxos: Vec<UtxoRecoveryData>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+struct UtxoRecoveryData {
+    /// Transaction ID as hex string
+    txid: String,
+    /// Output index
+    vout: u32,
+    /// Ethereum address (empty string for change UTXOs)
+    eth_address: String,
+}
+
+impl From<UtxoRecoveryData> for UtxoToRecover {
+    fn from(data: UtxoRecoveryData) -> Self {
+        UtxoToRecover {
+            outpoint: Some(OutPoint {
+                txid: hex::decode(&data.txid).unwrap_or_else(|_| {
+                    error!(target: "reth::cli", "Invalid txid hex: {}", data.txid);
+                    vec![0u8; 32] // Fallback to zeros
```

Is that a good decision? Consider implementing [`TryFrom`](https://doc.rust-lang.org/std/convert/trait.TryFrom.html) instead.

---

yeah you're right, better to throw an error than rely on btc-server rejecting the non existent utxo. I've had a go implementing `TryFrom` if you are able to take a look